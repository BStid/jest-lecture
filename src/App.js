import React, { Component } from 'react';
import Modal from './Modal';

class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showModal: false,
      isChecked: false
    };
  }

  toggleModal = () => this.setState({ showModal: !this.state.showModal });

  onChange = () => this.setState({ isChecked: !this.state.isChecked });

  render() {
    const { showModal } = this.state;

    return (
      <div>
        <button onClick={this.toggleModal}>Toggle Menu</button>
        {showModal ? (
          <Modal>
            <h1>Would you like to adopt a puppy?</h1>
            <div className="buttons">
              <button onClick={this.toggleModal}>Yes</button>
              <button onClick={this.toggleModal}>No</button>
            </div>
          </Modal>
        ) : null}

        <label>
          <input type="checkbox" checked={this.state.isChecked} onChange={this.onChange} />
          {this.state.isChecked ? this.props.labelOn : this.props.labelOff}
        </label>
      </div>
    );
  }
}

export default App;
