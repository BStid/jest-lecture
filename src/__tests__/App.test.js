import React from 'react';
import ReactDOM from 'react-dom';
import * as TestUtils from 'react-dom/test-utils';
import { create } from 'react-test-renderer';
import App from '../App';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('snapshot', () => {
  const c = create(<App />);
  expect(c.toJSON()).toMatchSnapshot();
});

test('shows modal when toggleModal is called', () => {
  const component = create(<App />);
  const instance = component.getInstance();

  expect(instance.state.showModal).toBe(false);
  instance.toggleModal();
  expect(instance.state.showModal).toBe(true);
});

it('CheckboxWithLabel changes the text after click', () => {
  // Render a checkbox with label in the document
  const checkbox = TestUtils.renderIntoDocument(<App labelOn="On" labelOff="Off" />);

  const checkboxNode = ReactDOM.findDOMNode(checkbox);

  // Verify that it's Off by default
  expect(checkboxNode.textContent).toEqual('Toggle MenuOff');

  // Simulate a click and verify that it is now On
  TestUtils.Simulate.change(TestUtils.findRenderedDOMComponentWithTag(checkbox, 'input'));
  expect(checkboxNode.textContent).toEqual('Toggle MenuOn');
});
