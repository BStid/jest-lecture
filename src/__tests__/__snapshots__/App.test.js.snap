// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Toggle Menu
  </button>
  <label>
    <input
      checked={false}
      onChange={[Function]}
      type="checkbox"
    />
  </label>
</div>
`;
